### PR TITLE
HBSP-459: add domain filter to page api

### PR DESCRIPTION
**Pull request by spotaichuk-smartling**

### CODE REVIEW

File: src/main/java/com/smartling/connector/hubspot/sdk/common/Next.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class Next {
+    private String link;
+    private String after;
+}
```
Comment (PavelLoparev) [inline src/main/java/com/smartling/connector/hubspot/sdk/common/Next.java:13]: Not used? Probably meant to be used in `Paging `?

File: src/main/java/com/smartling/connector/hubspot/sdk/v3/HubspotPagesV3Client.java [hunk 2]
```diff
@@ -26,7 +26,7 @@ public interface HubspotPagesV3Client extends HubspotClient
 
     String updatePageDraft(String page, String updatePageId) throws HubspotApiException;
 
-    ListWrapper<PageDetail> listPages(int offset, int limit, String orderBy, Map<String, Object> queryMap) throws HubspotApiException;
+    ListWrapper<PageDetail> listPages(int offset, int limit, String orderBy, String domain, Map<String, Object> queryMap) throws HubspotApiException;
```
Comment (sl-mmuradov) [inline src/main/java/com/smartling/connector/hubspot/sdk/v3/HubspotPagesV3Client.java:29]: There is no need to add `domain` filter here. We already have `queryMap` for all filters.

File: src/main/java/com/smartling/connector/hubspot/sdk/v3/rest/api/DomainsApi.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package com.smartling.connector.hubspot.sdk.v3.rest.api;
+
+import com.smartling.connector.hubspot.sdk.domain.DomainDetails;
+import feign.Param;
+import feign.RequestLine;
+
+public interface DomainsApi
+{
+    @RequestLine("GET /cms/v3/domains?limit={limit}&after={after}&sort={sort}")
+    DomainDetails listPages(@Param("limit") int limit, @Param("after") String after, @Param("sort") String sort);
```
Comment (sl-mmuradov) [inline src/main/java/com/smartling/connector/hubspot/sdk/v3/rest/api/DomainsApi.java:10]: Should be `listDomains` I believe.

File: src/main/java/com/smartling/connector/hubspot/sdk/v3/rest/api/PagesEntityApi.java [hunk 4]
```diff
@@ -17,6 +17,9 @@ public interface PagesEntityApi
     @RequestLine("GET /cms/v3/pages/{pageType}?limit={limit}&offset={offset}&sort={sortBy}&property=" + PageDetail.FIELDS)
     ListWrapper<PageDetail> listPages(@Param("pageType") String pageType, @Param("limit") int limit, @Param("offset") int offset, @Param("sortBy") String sortBy, @QueryMap Map<String, Object> queryMap);
 
+     @RequestLine("GET /cms/v3/pages/{pageType}?limit={limit}&offset={offset}&sort={sortBy}&domain={domain}&property=" + PageDetail.FIELDS)
```
Comment (sl-mmuradov) [inline src/main/java/com/smartling/connector/hubspot/sdk/v3/rest/api/PagesEntityApi.java:20]: You don't need extra this method, if `domain` parameter was put into `queryMap`.

File: src/test/java/com/smartling/connector/hubspot/sdk/v3/rest/HubspotRestPagesV3ClientTest.java [hunk 5]
```diff
@@ -210,24 +210,24 @@ public void shouldCallRefreshTokenFirst() throws HubspotApiException
...
+        hubspotClient.listPages(0, 10, null, null, null);
     }
 
     @Test
     public void shouldDeserializeFields() throws HubspotApiException
     {
         givenThat(get(HttpMockUtils.path("/cms/v3/pages/site-pages")).willReturn(HttpMockUtils.aJsonResponse(pageDetails())));
 
-        ListWrapper<PageDetail> pageDetails = hubspotClient.listPages(5, 15, null, null);
+        ListWrapper<PageDetail> pageDetails = hubspotClient.listPages(5, 15, null, null, null);
```
Comment (sl-mmuradov) [inline src/test/java/com/smartling/connector/hubspot/sdk/v3/rest/HubspotRestPagesV3ClientTest.java:230]: Don't forget to add a test for using filter by `domain`, please.

**Review by sl-mmuradov — APPROVED** [review]

**Pull request merged**